### PR TITLE
platform(general): trim API urls

### PR DESCRIPTION
**Pull request by mikeurbanski1**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally we encourage adding a scope to the prefix, which indicates the targeted framework, otherwise 'general' will be assumed.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Trims the BC and Prisma API URL inputs.
## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gruebel — APPROVED** [review]
I fixed the `mypy` issue for you 🙂

File: checkov/common/util/http_utils.py [hunk 1]
```diff
@@ -25,11 +25,27 @@
...
 
+@overload
+def normalize_bc_url(url: None) -> None:
+    ...
+
+
+@overload
+def normalize_bc_url(url: str) -> str:
+    ...
+
```
Comment (itai1357) [inline checkov/common/util/http_utils.py:36]: HI! I 
never saw such a using of overload.
can you please share why those 2 function are necessary?
it seem that the third one is enough

Comment (gruebel) [reply checkov/common/util/http_utils.py:36]: the `@overload` functionality in Python is only for typing purposes. It gives a possibility to differentiate between what kind of types go in and what to what should come out. In this example it is clear, if `None` goes in, `None` should come out, same for `str`.
If you don't add the `@overload` annotation, then the type checker will assume that either `str` or `None` could return, which can be ok, but then you have handle it downstream and always check, if it `is not None`

**Review by maxamel — APPROVED** [review]

**Review by itai1357 — APPROVED** [review]

File: checkov/common/util/http_utils.py (hunk 1 above)
Comment (itai1357) [reply checkov/common/util/http_utils.py:36]: cool!

**Pull request merged**